mlir/lib/Dialect/Math/Transforms/ExpandPatterns.cpp

```diff
@@ -660,6 +703,7 @@ void mlir::populateExpandExp2FPattern(RewritePatternSet &patterns) {
 }
 
 void mlir::populateExpandPowFPattern(RewritePatternSet &patterns) {
+  patterns.add(convertSpecialPowfOp);
```

Comment:
Question for MLIR experts: Here, we want the `convertSpecialPowfOp` to have precedence over the `convertPowfOp` pattern. Is that ensured by it being added first here? If not, do we need to merge these two patterns to ensure ordering?

Reply:
`patterns.add(convertSpecialPowfOp, /*benefit=*/ 2.0);`
This would explicitly give the order we want.

Reply:
I'm not very sure since I didn't check the code, but adding patterns in this order make `convertSpecialPowfOp` run first.

```cpp
patterns.add(convertSpecialPowfOp);
patterns.add(convertPowfOp);
```

Reply:
@hanhanW ?

Reply:
> I'm not very sure since I didn't check the code, but adding patterns in this order make convertSpecialPowfOp run first.

This is correct, but I think it is not documented. IMO, we prefer using `benefits` to prioritize the patterns.

https://mlir.llvm.org/docs/PatternRewriter/#benefit

Reply:
Updated to give explicit benefit=2!